src/code_transformation/differentiable.jl

```diff
@@ -67,6 +80,33 @@ function unionise_sig(code::Expr)
     return replace_body(code, new_body)
 end
 
+"""
+    unionise_struct(code)
+
+`code` should be an `Expr` containing the definition of a type. The type will only be
+changed if it is parametric. That is, if you wish to be able to differentiate through a
+user-defined type, it must contain only `Any`s and parametric types.
+"""
+function unionise_struct(code::Expr)
+    tmp = code.args[2]
+    name = tmp isa Expr && tmp.head == Symbol("<:") ? code.args[2].args[1] : code.args[2]
```

Comment:
Perhaps store `tmp isa Expr && tmp.head == Symbol("<:")` in a variable since you're reusing it two lines below.